cloudsim-plus-testbeds/src/main/java/org/cloudsimplus/sla/DynamicVmCreationByCpuUtilizationAndFreePesOfVm.java

```diff
@@ -201,16 +209,14 @@ private Vm selectVmForCloudlet(Cloudlet cloudlet) {
 
         Vm selectedVm = createdVms.stream()
                 .filter(vm -> getExpectedNumberOfFreeVmPes(vm, true) >= cloudlet.getNumberOfPes())
-                .filter(vm -> getExpectedCloudletResponseTime(cloudlet, vm) <= getResponseTimeSlaContract())
+                .filter(vm -> getExpectedCloudletResponseTime(cloudlet) <= responseTimeSlaContract)
                 .findFirst().orElse(mostFreePesVm);
 
         return selectedVm;
     }
 
-    private double getExpectedCloudletResponseTime(Cloudlet cloudlet, Vm vm) {
```

Comment:
The method should be as before, where it receives a Vm as parameter and gets the MIPS directly from it.
The real problem is: if you change your code so that VMs will have different MIPS capacity, using the vmMips attribute will make your simulation to wrongly use the same MIPS value for all VMs.